fix saving of black display names etc. of items in config

weaondara opened the pull request:
this fixes saving of items with black display names in yaml configs

kickash32 commented on Spigot-Server-Patches/0422-Fix-saving-of-black-display-names-etc.-of-items-in-c.patch:
```diff
@@ -0,0 +1,53 @@
++        return out.toString(); // Paper - fix deletion of color codes at the beginning of result string (old behavior deleted black color codes in display names of items)
+     }
+-
++    
```
Unnecessary diff

weaondara replied:
ok fixed that but it says branches are diverging
should i do a force push or is there another option
i used the 'paper edit server'

kickash32 replied:
force pushing to your branch should be fine

weaondara replied:
ok thx

weaondara commented on Spigot-Server-Patches/0422-Fix-saving-of-black-display-names-etc.-of-items-in-c.patch:
```diff
@@ -0,0 +1,53 @@
+             out.append(c.getText());
+         }
+-        return out.toString().replaceFirst("^(" + defaultColor + ")*", "");
++        return out.toString(); // Paper - fix deletion of color codes at the beginning of result string (old behavior deleted black color codes in display names of items)
```
where did i miss it?

kickash32 replied:
never mind I can't read...

weaondara committed:
Fix serialization of colors from components

This patch fixes the serialization of display names, item lores and
other things which use strings with color codes. The old implementation
deleted the color codes at the beginning of the resulting string if it
matched the default color passed to the conversion function. This
resulted in items having a black display name losing the black color
code in the beginning of the text when the item was serialized (e.g.
saving an ItemStack in a Yaml config).

Spigot has now made the issue worse and expanded the scope to more places.
